src/ophys_etl/transforms/mesoscope_2p/tiff.py

```diff
@@ -296,11 +309,22 @@ def is_multiscope(self):
 
     @property
     def plane_views(self):
-        self._planes = None
         if self._planes is None:
             self._planes = []
             page_offset = 0
+
+            z_has_been_processed = set()  # do not reprocess same z
+
             for z in self.plane_scans:
+
+                # relying on np.unique(self.plane_scans) forces
+                # us to process self.plane_scans in sorted order;
+                # nothing in the data model says that it is required
+                # that the plane scan z values be in sorted order
+                if z in z_has_been_processed:
+                    continue
+                z_has_been_processed.add(z)
```

Comment:
Trying to understand the purpose for the `if` statement? By definition a Python set can only contain unique elements and Python sets are unordered.

```
my_set = set()
my_set.add(1)
my_set.add(2)
my_set.add(1)
print(my_set)
```

Reply:
We only want to loop over unique values of z here (processing the same value of z twice was part of the bug that originally provoked this issue; sometimes the z values in the TIFF metadata get padded with multiple `0`s). Originally, this was achieved using `np.unique()`. Unfortunately, `np.unique` returns a sorted list, and we want to loop over the unique values of z in the order that they are stored in the TIFFs metadata.

Reply:
ah okay I see, I misread the code and thought that the end product variable was `z_has_been_processed`. 

You think you could also do something like:

```
_, idx = np.unique(self.plane_scans, return_index=True)
unique_ordered_zs = self.plane_scans[np.sort(idx)]
```

But if the number of `z`s in `self.plan_scans` is small, then the current solution is perfectly fine.